solidity/test/maintainer/Maintainer.test.ts

```diff
@@ -0,0 +1,3577 @@
+                            MultipleDepositsNoMainUtxo
+                          // Take wallet public key hash from first deposit. All
+                          // deposits in same sweep batch should have the same value
+                          // of that field.
+                          const { walletPubKeyHash } = data.deposits[0].reveal
+                          let sweepOutcome: ScenarioOutcome
+
+                          before(async () => {
+                            await createSnapshot()
+
+                            // Set the deposit dust threshold to 0.0001 BTC, i.e. 100x smaller than
+                            // the initial value in the Bridge in order to save test Bitcoins.
+                            await bridge.setDepositDustThreshold(10000)
+
+                            // Simulate the wallet is an Live one and is known in
+                            // the system.
+                            await bridge.setWallet(walletPubKeyHash, {
+                              ecdsaWalletID: ethers.constants.HashZero,
+                              mainUtxoHash: ethers.constants.HashZero,
+                              pendingRedemptionsValue: 0,
+                              createdAt: await lastBlockTime(),
+                              movingFundsRequestedAt: 0,
+                              closingStartedAt: 0,
+                              pendingMovedFundsSweepRequestsCount: 0,
+                              state: walletState.Live,
+                              movingFundsTargetWalletsCommitmentHash:
+                                ethers.constants.HashZero,
+                            })
+
+                            sweepOutcome = await runSweepScenario(data)
+                          })
+
+                          after(async () => {
+                            await restoreSnapshot()
+                          })
+
+                          it("should not revert", async () => {
+                            await expect(sweepOutcome.tx.wait()).not.to.be
+                              .reverted
+                          })
+
+                          it("should refund ETH", async () => {
+                            const postNotifyThirdPartyBalance =
+                              await provider.getBalance(thirdParty.address)
+                            const diff = postNotifyThirdPartyBalance.sub(
+                              sweepOutcome.initThirdPartyBalance
+                            )
+
+                            expect(diff).to.be.gt(0)
+                            expect(diff).to.be.lt(
+                              ethers.utils.parseUnits("1000000", "gwei") // 0,001 ETH
+                            )
+                          })
+                        }
+                      )
+                    })
+                  }
+                )
+              })
+            })
+          })
+        })
+      })
+    })
+
+    context("when the wallet state is MovingFunds", () => {
```

Comment:
I would drop this case, tbh. There is no difference in processing for Live and MovingFunds wallet. What we need to cover are the differences in input data and whether deposits are routed to vaults or not.

Reply:
Removed